osu.Game/Overlays/Chat/ChannelList/ChannelListItem.cs

```diff
@@ -158,9 +149,37 @@ private Drawable createIcon()
             };
         }
 
+        private ChannelListItemMentionPill? createMentionPill()
+        {
+            if (isSelector)
+                return null;
+
+            return new ChannelListItemMentionPill
+            {
+                Anchor = Anchor.CentreLeft,
+                Origin = Anchor.CentreLeft,
+                Margin = new MarginPadding { Right = 3 },
+                Mentions = { BindTarget = Mentions },
+            };
+        }
+
+        private ChannelListItemCloseButton? createCloseButton()
+        {
+            if (isSelector)
+                return null;
+
+            return new ChannelListItemCloseButton
+            {
+                Anchor = Anchor.CentreLeft,
+                Origin = Anchor.CentreLeft,
+                Margin = new MarginPadding { Right = 3 },
+                Action = () => OnRequestLeave?.Invoke(Channel),
+            };
+        }
+
         private void updateState()
         {
-            bool selected = selectedChannel.Value == Channel;
+            bool selected = selectedChannel.Value == Channel || (isSelector && selectedChannel.Value == null);
```

Comment:
I really dislike this null check. Why is it required? Why can't we be assured that the selected channel is always the one specific to the channel list?

Reply:
In cases where the user leaves the current channel, the channel manager will set the current channel to null, which opens the selector as specified in this condition: https://github.com/ppy/osu/blob/136ecb45e241b6d0dd4358107ddb2d9b4299f3bf/osu.Game/Overlays/ChatOverlayV2.cs#L247

This null check matches that so the selector list item appears selected correctly.

Would a better solution be to automatically set the current channel to the `ChannleListingChannel` whenever it becomes null?

Reply:
Yes, I believe the `ChatOverlay` should set the channel to the `ChannelListingChange` when it sees the channel become `null`.